HW1/P2/P2.py

```diff
@@ -12,20 +20,36 @@ def mandelbrot(x, y):
     return iteration
 
 def sum_values_for_partitions(rdd):
-    'Returns (as an RDD) the sum of V for each partition of a (K, V) RDD'
+ #   'Returns (as an RDD) the sum of V for each partition of a (K, V) RDD'
     # note that the function passed to mapPartitions should return a sequence,
     # not a value.
     return rdd.mapPartitions(lambda part: [sum(V for K, V in part)])
 
 def draw_image(rdd):
-    '''Given a (K, V) RDD with K = (I, J) and V = count,
-    display an image of count at each I, J'''
-
+#    '''Given a (K, V) RDD with K = (I, J) and V = count,
+#    display an image of count at each I, J'''
     data = rdd.collect()
+    #
     I = np.array([d[0][0] for d in data])
     J = np.array([d[0][1] for d in data])
     C = np.array([d[1] for d in data])
+    #
     im = np.zeros((I.max() + 1, J.max() + 1))
     im[I, J] = np.log(C + 1)  # log intensity makes it easier to see levels
     plt.imshow(im, cmap=cm.gray)
     plt.show()
+    plt.savefig("Mandelbrot", dpi=2000, facecolor='w', edgecolor='w')
+
+q = range(0,4000)
+X = sc.parallelize(q,100)
+coordinates = X.flatMap(lambda x: [(x,y) for y in q]) # successfully creates pixel coordinates...
```

Comment:
Thanks! That's the most elegant solution for getting the grid coordinates I've seen.